### PR TITLE
feat: map logger level to OTel level

### PR DESCRIPTION
**Pull request by khushijain21**

### CODE REVIEW

File: instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb [hunk 1]
```diff
@@ -38,10 +37,20 @@ def skip_instrumenting?
           end
 
           def severity_number(severity)
-            ::Logger::Severity.const_get(severity)
-          rescue NameError => e
-            OpenTelemetry.logger.warn(message: "Unable to coerce severity text #{severity} into severity_number. Setting severity_number to nil.", exception: e)
-            nil
+            case severity.downcase
+            when 'debug'
+              Opentelemetry::Proto::Logs::V1::SeverityNumber::SEVERITY_NUMBER_DEBUG
```
Comment (kaylareopelle) [inline instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb]: I believe these constants aren't available in any of the current dependencies for this gem. Right now, they're part of the OTLP exporters. I don't think we want to require an exporter for some instrumentation. 

One potential workaround would be to update the draft API to have a set of constants mapping to these proto values. However, I'm sure there's other ways to solve this problem and I'd love to hear your ideas.

File: .toys/.data/releases.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ gems:
 
   - name: opentelemetry-instrumentation-logger
     directory: instrumentation/logger
-    version_constant: [OpenTelemetry, Instrumentation, Logger, VERSION
+    version_constant: [OpenTelemetry, Instrumentation, Logger, VERSION]
```
Comment (kaylareopelle) [inline .toys/.data/releases.yml:52]: Thanks for catching this!

**Review by kaylareopelle — COMMENTED** [review]
Thanks for this PR, @khushijain21! 

* To get the CI to run, could you update the title of your PR to have the format of a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)? This is a requirement in the main `contrib` repo that was added after I created the initial PR.

* I created a PR to help allow the tests to run: https://github.com/khushijain21/opentelemetry-ruby-contrib/pull/1 
They also incorporate the latest version of the draft logs SDK, which causes a few failures unrelated to your changes. Let me know if you'd like to address those, or if you prefer I take care of them. If the PR doesn't allow you to run the tests, let me know and we can work together to find another solution.

File: instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb (hunk 1 above)
Comment (khushijain21) [reply instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb]: yes, I think having a map of these in draft API would make it accessible to all bridges!

**Review by kaylareopelle — COMMENTED** [review]
@khushijain21 these updates look good! Thank you! Once the CI passes, I'll approve and merge.

**Review by kaylareopelle — APPROVED** [review]
I'm not sure why the full suite didn't run, but I feel confident about putting this change into the feature branch. We can deal with any follow-ups later. 

Thank you!

**Pull request merged**